### PR TITLE
Implements MGET command

### PR DESCRIPTION
**Pull request by bsbodden**
See https://oss.redislabs.com/redisjson/commands/#jsonmget

### CODE REVIEW

**Review by sazzad16 — CHANGES_REQUESTED** [review]
Format with spaces.

**Comment by bsbodden** [comment]
> Format with spaces.

That's what I get for opening this with Eclipse. I took care of it.

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 1]
```diff
@@ -217,6 +221,46 @@ public Long del(String key, Path path) {
     	return gson.fromJson(rep, clazz);
     }
 
+    /**
+     * Returns the documents from multiple key s. Non-existing keys are reported as
+     * null.
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:226]: `key s`?

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 2]
```diff
@@ -217,6 +221,46 @@ public Long del(String key, Path path) {
...
+     * @param keys  keys for the JSON documents
+     * @return a List of documents rooted at path
+     */
+    public <T> List<T> mget(Class<T> clazz, String... keys) {
+        return mget(Path.ROOT_PATH, clazz, keys);
+    }
+
+    /**
+     * Returns the values at path from multiple key s. Non-existing keys and
+     * non-existing paths are reported as null.
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:239]: `key s`?

File: src/test/java/com/redislabs/modules/rejson/ClientTest.java [hunk 3]
```diff
@@ -196,4 +256,81 @@ public void type1Exception() throws Exception {
...
+        Baz baz2 = new Baz("quuz2", "grault2", "waldo2");
+        Qux qux1 = new Qux("quux1", "corge1", "garply1", baz1);
+        Qux qux2 = new Qux("quux2", "corge2", "garply2", baz2);
+
+        client.set("qux1", qux1);
+        client.set("qux2", qux2);
+
+        List<Qux> allQux = client.mget(Qux.class, "qux1", "qux2", "qux3");
+
+        System.out.println(allQux);
```
Comment (sazzad16) [inline src/test/java/com/redislabs/modules/rejson/ClientTest.java:303]: Use a logging api if necessary. `slf4j` could be available.
```suggestion
```

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java [hunk 4]
```diff
@@ -217,6 +221,46 @@ public Long del(String key, Path path) {
...
+     * @param keys  keys for the JSON documents
+     * @return a List of documents rooted at path
+     */
+    public <T> List<T> mget(Path path, Class<T> clazz, String... keys) {
+        String[] args = Stream //
+                .of(keys, new String[] { path.toString() }) //
+                .flatMap(Stream::of) //
+                .toArray(String[]::new);
+
+        List<String> rep = Collections.emptyList();
```
Comment (sazzad16) [inline src/main/java/com/redislabs/modules/rejson/JReJSON.java:253]: This should be okay.
```suggestion
        List<String> rep;
```

File: src/test/java/com/redislabs/modules/rejson/ClientTest.java [hunk 5]
```diff
@@ -196,4 +256,81 @@ public void type1Exception() throws Exception {
...
+
+        assertEquals(3, allQux.size());
+        allQux.removeAll(Collections.singleton(null));
+        assertEquals(2, allQux.size());
```
Comment (sazzad16) [inline src/test/java/com/redislabs/modules/rejson/ClientTest.java:307]: Shouldn't `null` be the third element? Can we assert that as well?

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 1 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: fixed

File: src/main/java/com/redislabs/modules/rejson/JReJSON.java (hunk 2 above)
Comment (bsbodden) [reply src/main/java/com/redislabs/modules/rejson/JReJSON.java]: fixed the typo

File: src/test/java/com/redislabs/modules/rejson/ClientTest.java (hunk 3 above)
Comment (bsbodden) [reply src/test/java/com/redislabs/modules/rejson/ClientTest.java]: removed, left overs when debugging

File: src/test/java/com/redislabs/modules/rejson/ClientTest.java (hunk 5 above)
Comment (bsbodden) [reply src/test/java/com/redislabs/modules/rejson/ClientTest.java:331]: Good idea

File: src/test/java/com/redislabs/modules/rejson/ClientTest.java [hunk 6]
```diff
@@ -223,6 +280,81 @@ public void typeException() throws Exception {
...
+        Qux qux1 = new Qux("quux1", "corge1", "garply1", baz1);
+        Qux qux2 = new Qux("quux2", "corge2", "garply2", baz2);
+
+        client.set("qux1", qux1);
+        client.set("qux2", qux2);
+
+        List<Qux> allQux = client.mget(Qux.class, "qux1", "qux2", "qux3");
+
+        assertEquals(3, allQux.size());
+        assertEquals(null, allQux.get(2));
```
Comment (sazzad16) [inline src/test/java/com/redislabs/modules/rejson/ClientTest.java:329]:
```suggestion
        assertNull(allQux.get(2));
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]
Don't get annoyed :P

File: src/test/java/com/redislabs/modules/rejson/ClientTest.java (hunk 6 above)
Comment (bsbodden) [reply src/test/java/com/redislabs/modules/rejson/ClientTest.java:329]: Not at all! Good catch!

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**